src/javaserver/Handlers/ConnectionHandler.java

```diff
@@ -30,9 +33,11 @@ public void run() {
 
             System.out.println(requestString);
             HttpRequestParser requestParser = new HttpRequestParser(requestString);
-            HttpResponseHandler responseHandler = new HttpResponseHandler(requestParser);
+            logger.addLog(requestParser.statusCode());
+            RequestHandler requestHandler = new RequestHandler(requestParser, logger);
```

Comment:
This `run` method is getting a bit long. At the very least, maybe some methods could be extracted.